resize-observer/devicepixel2.html

```diff
@@ -0,0 +1,157 @@
+  }
+  .outer {
+    width: 99.5px;
+    display: flex;
+  }
+  .outer>* {
+    flex: 1 1 auto;
+    height: 20px;
+  }
+</style>
+<body>
+  <div id="back"></div>
+  <div id="front"></div>
+</body>
+<script>
+const create2x2CheckerboardDataURL = function() {
+  const ctx = document.createElement('canvas').getContext('2d');
+  ctx.canvas.width = 2;
+  ctx.canvas.height = 2;
+
+  return function create2x2CheckerboardDataURL() {
+    ctx.fillRect(0, 0, 1, 1);
+    ctx.fillRect(1, 1, 1, 1);
+    return ctx.canvas.toDataURL();
+  };
+}();
+const patternURL = create2x2CheckerboardDataURL();
+
+function setPattern(elem, x, y, width, height) {
+  const oneDevicePixel = 1 / devicePixelRatio;
+  const twoDevicePixels = oneDevicePixel * 2;
+  const xOff = x % 2;
+  const yOff = y % 2;
+  elem.style.backgroundImage = `url("${patternURL}")`;
+  elem.style.backgroundSize = `${twoDevicePixels}px ${twoDevicePixels}px`;
+  elem.style.backgroundPosition = `${xOff * oneDevicePixel}px ${yOff * oneDevicePixel}px`;
+}
+
+/*
+This test creates elements like this
+
+   <div id="back"></div>
+   <div id="front">
+     <div class="outer"><div></div><div></div></div>
+     <div class="outer"><div></div><div></div><div></div></div>
+     <div class="outer"><div></div><div></div><div></div><div></div></div>
+     ...
+   </div>
+
+It has both #front and #back starting at the top left corner of the page
+so they should both have a devicePixel position of 0,0
+
+It then sets a repeating background-image and background-size on the the
+#back element and all of the .outer children elements so they have a 2x2
+checkerboard image at a size that should draw each pixel of the 2x2 
+checkerboard pattern in devicePixels.
+
+The devicePixelContentBoxSize of all the elements is queried with
+a ResizeObserver.
+
+Finally the background-position of each child element is set so that
+the patterns should align.
+
+In other words
+
+  <div class="outer">
+    <div></div><div></div><div></div>
+  </div>
+
+The first child will be displayed at left = 0 so it background-position is
+set to 0px 0px. The second child should be displayed at 
+
+secondChildLeft = entry(firstChild).devicePixelContentBoxSize[0].inlineSize. 
+
+The 3rd child should be displayed at 
+
+thirdChildLeft = entry(firstChild).devicePixelContentBoxSize[0].inlineSize + 
+                 entry(secondChild).devicePixelContentBoxSize[0].inlineSize
+
+If the element's left edge is an odd value then we need to set the
+background-position of the element's pattern +1 devicePixel.
+
+If what devicePixelContentBox reports is correct the children's patterns
+will align with the pattern in #back and the children will be invisible.
+If what devicePixelContentBox reports is incorrect the patterns will not
+align and it should be clearly visible.
+*/
+
+const frontElem = document.querySelector('#front');
+const backElem = document.querySelector('#back');
+
+const elemToEntry = new Map();
+
+function setPatternInEntries(entries) {
```

Comment:
"setPatternInEntries" is a bit of a confusing name, since we're not putting the pattern in the entries.

Maybe rename this to "setPatternsUsingSizeInfo" or something along those lines?  That seems to be what this does, rather than doing something "in the entries".

Reply:
done